In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## TS000153262

## Análisis de geocercas en todas las unidades.

In [2]:
montero_gio=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/Geofences InOut Report.csv')
montero_gio.info()
montero_gio.sort_values(by=['EventTime'], ascending=False)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4006 entries, 0 to 4005
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   textbox12       4006 non-null   object 
 1   UnitName        4006 non-null   object 
 2   textbox18       0 non-null      float64
 3   EventTime       4006 non-null   object 
 4   GeofenceName    4006 non-null   object 
 5   GeofenceEvent1  4006 non-null   object 
 6   Schedule        0 non-null      float64
dtypes: float64(2), object(5)
memory usage: 219.2+ KB


textbox12     UnitName  textbox18                 EventTime  \
2946   Group:   Unit:WHI947        NaN   20/05/2025 9:58:14 a.m.   
2945   Group:   Unit:WHI947        NaN   20/05/2025 9:57:14 a.m.   
325    Group:   Unit:SPH808        NaN   20/05/2025 9:54:37 a.m.   
3914   Group:   Unit:UAO056        NaN   20/05/2025 9:44:57 a.m.   
2336   Group:   Unit:UAN646        NaN   20/05/2025 9:43:52 a.m.   
...        ...          ...        ...                       ...   
2685   Group:   Unit:TKI593        NaN  12/05/2025 10:05:48 a.m.   
1622   Group:   Unit:UAN790        NaN  12/05/2025 10:01:17 a.m.   
2463   Group:   Unit:UAO890        NaN  12/05/2025 10:01:08 a.m.   
1621   Group:   Unit:UAN790        NaN  12/05/2025 10:00:27 a.m.   
2462   Group:   Unit:UAO890        NaN  12/05/2025 10:00:08 a.m.   

                                           GeofenceName GeofenceEvent1  \
2946        2.- Reloj Rodeo : cartagena rodeo cartagena            OUT   
2945        2.- Reloj Rodeo : cartagena rodeo cartagena             IN   
325   4.- Tienda El Tope : cartagena sanfernando car...            OUT   
3914  1. Turno de despacho : Calle 34 # 17 231 Carta...             IN   
2336  4.- Tienda El Tope : cartagena sanfernando car...             IN   
...                                                 ...            ...   
2685  4.- Tienda El Tope : cartagena sanfernando car...             IN   
1622  5.,- Entrada a Blas de Lezo : cartagena blasde...            OUT   
2463  4.- Tienda El Tope : cartagena sanfernando car...            OUT   
1621  5.,- Entrada a Blas de Lezo : cartagena blasde...             IN   
2462  4.- Tienda El Tope : cartagena sanfernando car...             IN   

      Schedule  
2946       NaN  
2945       NaN  
325        NaN  
3914       NaN  
2336       NaN  
...        ...  
2685       NaN  
1622       NaN  
2463       NaN  
1621       NaN  
2462       NaN  

[4006 rows x 7 columns]

In [3]:
unit_name=montero_gio['UnitName']
unit_name_conts=unit_name.value_counts()
unit_name_conts

UnitName
Unit:XVU383     193
Unit:XVO358     185
Unit:SLP174     176
Unit:XMB461     169
Unit:UAN790     153
Unit:XVN243     150
Unit:XVO410     147
Unit:XVP649     146
Unit:XVO853     145
Unit:UAR598     143
Unit:TPQ091     141
Unit:UAP929     141
Unit:TKI593     140
Unit:WHI947     136
Unit:UAO041     134
Unit:SJR801     117
Unit:SRR661     117
Unit:UAO056     117
Unit:SRR902     115
Unit:TNF678     115
Unit:UAO940     113
Unit:UAN646     107
Unit:TPY950     106
Unit:TPR 078    103
Unit:UAO890     103
Unit:SPH808      99
Unit:UAO861      95
Unit:UAM325      93
Unit:SLP442      92
Unit:XVO856      83
Unit:UAR494      74
Unit:UAQ054      38
Unit:TKI052      20
Name: count, dtype: int64

In [4]:
geo_fences=montero_gio['GeofenceName']
geo_fences_conts=geo_fences.value_counts()
geo_fences_conts

GeofenceName
4.- Tienda El Tope : cartagena sanfernando cartagena                                                 1036
1. Turno de despacho : Calle 34 # 17 231 Cartagena                                                    605
5.,- Entrada a Blas de Lezo : cartagena blasdelezo cartagena                                          356
7.- Antiguo Peaje Manga : Cartagena manga cartagena                                                   352
9.- Olimpica Bocagrande Cra. 3º : cartagena bocagrande cartagena                                      340
8.- Parque Espiritu del Manglar : cartagena centro cartagena                                          302
2.- Reloj Rodeo : cartagena rodeo cartagena                                                           260
12.- Estbas del Bosque. : cartagena bosque cartagena                                                  223
6.- Seguro Social : cartagena traversal 54 cartagena                                                  181
11.- Avenida del Lago : cartagena

## UAR598

R00A12V03, 0102

In [5]:
uar_598=montero_gio[montero_gio['UnitName'].str.contains('598')]
uar_598_16_05=uar_598[uar_598['EventTime'].str.contains('18/05')]
uar_598_16_05


textbox12     UnitName  textbox18                 EventTime  \
583   Group:   Unit:UAR598        NaN   18/05/2025 7:31:27 a.m.   
584   Group:   Unit:UAR598        NaN   18/05/2025 7:31:28 a.m.   
585   Group:   Unit:UAR598        NaN   18/05/2025 7:32:12 a.m.   
586   Group:   Unit:UAR598        NaN   18/05/2025 7:33:28 a.m.   
587   Group:   Unit:UAR598        NaN   18/05/2025 7:34:28 a.m.   
588   Group:   Unit:UAR598        NaN   18/05/2025 8:55:34 a.m.   
589   Group:   Unit:UAR598        NaN  18/05/2025 10:45:22 a.m.   
590   Group:   Unit:UAR598        NaN  18/05/2025 11:28:23 a.m.   
591   Group:   Unit:UAR598        NaN  18/05/2025 11:28:48 a.m.   
592   Group:   Unit:UAR598        NaN  18/05/2025 11:56:19 a.m.   
593   Group:   Unit:UAR598        NaN  18/05/2025 11:57:19 a.m.   
594   Group:   Unit:UAR598        NaN  18/05/2025 12:57:45 p.m.   
595   Group:   Unit:UAR598        NaN   18/05/2025 1:01:20 p.m.   
596   Group:   Unit:UAR598        NaN   18/05/2025 2:40:26 p.m.   
597   Group:   Unit:UAR598        NaN   18/05/2025 3:13:47 p.m.   
598   Group:   Unit:UAR598        NaN   18/05/2025 3:51:28 p.m.   
599   Group:   Unit:UAR598        NaN   18/05/2025 5:18:31 p.m.   
600   Group:   Unit:UAR598        NaN   18/05/2025 5:37:04 p.m.   

                                          GeofenceName GeofenceEvent1  \
583  4.- Tienda El Tope : cartagena sanfernando car...             IN   
584  4.- Tienda El Tope : cartagena sanfernando car...            OUT   
585  4.- Tienda El Tope : cartagena sanfernando car...             IN   
586  4.- Tienda El Tope : cartagena sanfernando car...             IN   
587  4.- Tienda El Tope : cartagena sanfernando car...            OUT   
588  10.- Clinica San Jose de Torices : cartagena b...             IN   
589  4.- Tienda El Tope : cartagena sanfernando car...             IN   
590  8.- Parque Espiritu del Manglar : cartagena ce...             IN   
591  8.- Parque Espiritu del Manglar : cartagena ce...            OUT   
592  9.- Olimpica Bocagrande Cra. 3º : cartagena bo...             IN   
593  9.- Olimpica Bocagrande Cra. 3º : cartagena bo...            OUT   
594  4.- Tienda El Tope : cartagena sanfernando car...             IN   
595  4.- Tienda El Tope : cartagena sanfernando car...            OUT   
596  4.- Tienda El Tope : cartagena sanfernando car...             IN   
597  7.- Antiguo Peaje Manga : Cartagena manga cart...            OUT   
598  9.- Olimpica Bocagrande Cra. 3º : cartagena bo...             IN   
599  1. Turno de despacho : Calle 34 # 17 231 Carta...             IN   
600  1. Turno de despacho : Calle 34 # 17 231 Carta...            OUT   

     Schedule  
583       NaN  
584       NaN  
585       NaN  
586       NaN  
587       NaN  
588       NaN  
589       NaN  
590       NaN  
591       NaN  
592       NaN  
593       NaN  
594       NaN  
595       NaN  
596       NaN  
597       NaN  
598       NaN  
599       NaN  
600       NaN

In [6]:
uar_598_geo_fences=uar_598['GeofenceName']
uar_598_geo_fences_conts=uar_598_geo_fences.value_counts()
uar_598_geo_fences_conts

GeofenceName
4.- Tienda El Tope : cartagena sanfernando cartagena                    40
5.,- Entrada a Blas de Lezo : cartagena blasdelezo cartagena            18
1. Turno de despacho : Calle 34 # 17 231 Cartagena                      17
7.- Antiguo Peaje Manga : Cartagena manga cartagena                     17
9.- Olimpica Bocagrande Cra. 3º : cartagena bocagrande cartagena        14
12.- Estbas del Bosque. : cartagena bosque cartagena                     8
8.- Parque Espiritu del Manglar : cartagena centro cartagena             8
2.- Reloj Rodeo : cartagena rodeo cartagena                              6
6.- Seguro Social : cartagena traversal 54 cartagena                     4
prueba mon                                                               3
3.- Olimpica San Fernando : cartagena sanfernando Cartagena              3
10.- Clinica San Jose de Torices : cartagena barrio torice cartagena     2
5.,- Entrada a Blas de Lezo : Dg21 45-92, Cartagena, Bolivar             1
8.- Parque E

## Análisis después de configurar la geocerca. Incialmente se hizo circular para GEOID=0, pero se actualizó a poligonal el 15-05-2025. No funcionó la geocerca poligonal, volvió a crearse la circular 17-05-2025.

In [7]:
raw_data=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/Export.xlsx - Sheet1.csv')
raw_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5116 entries, 0 to 5115
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  5116 non-null   object
 1   Message  5116 non-null   object
 2   Status   5116 non-null   object
dtypes: object(3)
memory usage: 120.0+ KB


In [8]:
resp_gtgeo=raw_data[raw_data['Message'].str.contains('RESP:GTGEO')]
resp_gtgeo.to_csv('/home/jlhb1984/py-projects/data/geofences_on_device/resp_gtgeo.csv')
resp_gtgeo

Created  \
15    2025-05-18 00:10:11.945   
16    2025-05-18 00:10:35.656   
31    2025-05-18 00:21:35.940   
32    2025-05-18 00:21:56.842   
64    2025-05-18 00:38:02.911   
...                       ...   
5050  2025-05-21 16:33:08.349   
5052  2025-05-21 16:33:17.235   
5090  2025-05-21 16:53:33.395   
5094  2025-05-21 16:54:57.871   
5095  2025-05-21 16:55:09.264   

                                                Message  \
15    +RESP:GTGEO,500700,868574042858816,,,11,1,1,10...   
16    +RESP:GTGEO,500700,868574042858816,,,10,1,1,14...   
31    +RESP:GTGEO,500700,868574042858816,,,C1,1,1,19...   
32    +RESP:GTGEO,500700,868574042858816,,,C0,1,1,21...   
64    +RESP:GTGEO,500700,868574042858816,,,81,1,1,36...   
...                                                 ...   
5050  +RESP:GTGEO,500700,868574042858816,,,31,1,1,24...   
5052  +RESP:GTGEO,500700,868574042858816,,,30,1,1,34...   
5090  +RESP:GTGEO,500700,868574042858816,,,01,1,1,13...   
5094  +RESP:GTGEO,500700,868574042858816,,,00,1,1,4....   
5095  +RESP:GTGEO,500700,868574042858816,,,01,1,1,0....   

                                                 Status  
15    0x2B524553503A475447454F2C3530303730302C383638...  
16    0x2B524553503A475447454F2C3530303730302C383638...  
31    0x2B524553503A475447454F2C3530303730302C383638...  
32    0x2B524553503A475447454F2C3530303730302C383638...  
64    0x2B524553503A475447454F2C3530303730302C383638...  
...                                                 ...  
5050  0x2B524553503A475447454F2C3530303730302C383638...  
5052  0x2B524553503A475447454F2C3530303730302C383638...  
5090  0x2B524553503A475447454F2C3530303730302C383638...  
5094  0x2B524553503A475447454F2C3530303730302C383638...  
5095  0x2B524553503A475447454F2C3530303730302C383638...  

[334 rows x 3 columns]

In [9]:
resp_gtpeo=raw_data[raw_data['Message'].str.contains('ACK')]
resp_gtpeo

Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [10]:
resp_gtgin=raw_data[raw_data['Message'].str.contains('GTGIN')]
resp_gtgin

Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [11]:
resp_gtgot=raw_data[raw_data['Message'].str.contains('GTGOT')]
resp_gtgot

Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [12]:
resp_gtfri=raw_data[raw_data['Message'].str.contains('GTFRI')]
resp_gtfri

Created  \
0     2025-05-18 00:00:05.768   
1     2025-05-18 00:01:05.159   
2     2025-05-18 00:02:05.249   
3     2025-05-18 00:03:06.509   
4     2025-05-18 00:03:20.953   
...                       ...   
5111  2025-05-21 17:05:16.748   
5112  2025-05-21 17:06:16.770   
5113  2025-05-21 17:07:16.805   
5114  2025-05-21 17:08:16.810   
5115  2025-05-21 17:09:16.813   

                                                Message  \
0     +RESP:GTFRI,500700,868574042858816,,,10,1,1,25...   
1     +RESP:GTFRI,500700,868574042858816,,,10,1,1,21...   
2     +RESP:GTFRI,500700,868574042858816,,,10,1,1,21...   
3     +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   
4     +RESP:GTFRI,500700,868574042858816,,,11,1,1,18...   
...                                                 ...   
5111  +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   
5112  +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   
5113  +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   
5114  +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   
5115  +RESP:GTFRI,500700,868574042858816,,,10,1,1,0....   

                                                 Status  
0     0x2B524553503A47544652492C3530303730302C383638...  
1     0x2B524553503A47544652492C3530303730302C383638...  
2     0x2B524553503A47544652492C3530303730302C383638...  
3     0x2B524553503A47544652492C3530303730302C383638...  
4     0x2B524553503A47544652492C3530303730302C383638...  
...                                                 ...  
5111  0x2B524553503A47544652492C3530303730302C383638...  
5112  0x2B524553503A47544652492C3530303730302C383638...  
5113  0x2B524553503A47544652492C3530303730302C383638...  
5114  0x2B524553503A47544652492C3530303730302C383638...  
5115  0x2B524553503A47544652492C3530303730302C383638...  

[4297 rows x 3 columns]

## TPR078

In [13]:
tpr078=montero_gio[montero_gio['UnitName'].str.contains('078')]
#tpr078_17_05=tpr078[tpr078['EventTime'].str.contains('18/05')]
tpr078

textbox12      UnitName  textbox18                EventTime  \
937    Group:   Unit:TPR 078        NaN  15/05/2025 5:10:40 a.m.   
938    Group:   Unit:TPR 078        NaN  15/05/2025 5:11:05 a.m.   
939    Group:   Unit:TPR 078        NaN  15/05/2025 5:55:46 a.m.   
940    Group:   Unit:TPR 078        NaN  15/05/2025 5:56:46 a.m.   
941    Group:   Unit:TPR 078        NaN  15/05/2025 7:42:08 a.m.   
...        ...           ...        ...                      ...   
1035   Group:   Unit:TPR 078        NaN  20/05/2025 3:24:11 p.m.   
1036   Group:   Unit:TPR 078        NaN  20/05/2025 3:24:57 p.m.   
1037   Group:   Unit:TPR 078        NaN  20/05/2025 4:20:12 p.m.   
1038   Group:   Unit:TPR 078        NaN  20/05/2025 4:21:13 p.m.   
1039   Group:   Unit:TPR 078        NaN  20/05/2025 4:23:13 p.m.   

                                           GeofenceName GeofenceEvent1  \
937   12.- Estbas del Bosque. : cartagena bosque car...             IN   
938   12.- Estbas del Bosque. : cartagena bosque car...            OUT   
939   4.- Tienda El Tope : cartagena sanfernando car...             IN   
940   4.- Tienda El Tope : cartagena sanfernando car...            OUT   
941         2.- Reloj Rodeo : cartagena rodeo cartagena             IN   
...                                                 ...            ...   
1035  5.,- Entrada a Blas de Lezo : cartagena blasde...             IN   
1036  5.,- Entrada a Blas de Lezo : cartagena blasde...            OUT   
1037  1. Turno de despacho : Calle 34 # 17 231 Carta...             IN   
1038  1. Turno de despacho : Calle 34 # 17 231 Carta...             IN   
1039  1. Turno de despacho : Calle 34 # 17 231 Carta...            OUT   

      Schedule  
937        NaN  
938        NaN  
939        NaN  
940        NaN  
941        NaN  
...        ...  
1035       NaN  
1036       NaN  
1037       NaN  
1038       NaN  
1039       NaN  

[103 rows x 7 columns]

In [14]:
tpr078_geo_fences=tpr078['GeofenceName']
tpr078_geo_fences_conts=tpr078_geo_fences.value_counts()
tpr078_geo_fences_conts

GeofenceName
4.- Tienda El Tope : cartagena sanfernando cartagena                27
1. Turno de despacho : Calle 34 # 17 231 Cartagena                  17
2.- Reloj Rodeo : cartagena rodeo cartagena                         13
5.,- Entrada a Blas de Lezo : cartagena blasdelezo cartagena        11
11.- Avenida del Lago : cartagena avenida el lago cartagena         10
12.- Estbas del Bosque. : cartagena bosque cartagena                 8
9.- Olimpica Bocagrande Cra. 3º : cartagena bocagrande cartagena     8
8.- Parque Espiritu del Manglar : cartagena centro cartagena         7
7.- Antiguo Peaje Manga : Cartagena manga cartagena                  2
Name: count, dtype: int64

## Hoy 17-05-2025 se envió comando para geocerca poligonal 1. Turno de despacho

In [15]:
raw_data_TPR078=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/Export.xlsx - Sheet2.csv')
raw_data_TPR078.info()
raw_data_TPR078_GTGEO=raw_data_TPR078[raw_data_TPR078['Message'].str.contains('RESP:GTGEO')]
raw_data_TPR078_GTGEO

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17396 entries, 0 to 17395
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  17396 non-null  object
 1   Message  17396 non-null  object
 2   Status   17396 non-null  object
dtypes: object(3)
memory usage: 407.8+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [16]:
resp_gtgeo_=raw_data_TPR078[raw_data_TPR078['Message'].str.contains('GTPEO')]
#resp_gtgeo_TPR078.to_csv('/content/drive/MyDrive/TSO Mobile/Colab/Geofence on device/resp_gtgeo.csv')
resp_gtgeo_

Created  \
1224  2025-05-17 21:19:13.050   
2323  2025-05-19 15:20:06.353   
2330  2025-05-19 15:24:41.875   
2371  2025-05-19 15:47:52.096   
4441  2025-05-20 22:36:55.323   
4443  2025-05-20 22:37:14.948   
4450  2025-05-20 22:41:21.048   

                                                Message  \
1224  +ACK:GTPEO,500700,868574042857875,,0,FFFF,2025...   
2323  +ACK:GTPEO,500700,868574042857875,,1,FFFF,2025...   
2330  +ACK:GTPEO,500700,868574042857875,,1,FFFF,2025...   
2371  +ACK:GTPEO,500700,868574042857875,,1,FFFF,2025...   
4441  +ACK:GTPEO,500700,868574042857875,,0,FFFF,2025...   
4443  +ACK:GTPEO,500700,868574042857875,,1,FFFF,2025...   
4450  +ACK:GTPEO,500700,868574042857875,,0,FFFF,2025...   

                                                 Status  
1224  0x2B41434B3A475450454F2C3530303730302C38363835...  
2323  0x2B41434B3A475450454F2C3530303730302C38363835...  
2330  0x2B41434B3A475450454F2C3530303730302C38363835...  
2371  0x2B41434B3A475450454F2C3530303730302C38363835...  
4441  0x2B41434B3A475450454F2C3530303730302C38363835...  
4443  0x2B41434B3A475450454F2C3530303730302C38363835...  
4450  0x2B41434B3A475450454F2C3530303730302C38363835...

Se procedió a crear una geocerca poligonal de 4 coordenadas cerca a Tienda El Tope.

Comando para borrar coordenadas en geocerca 0:
AT+GTPEO=gv55w,0,0,1,10,,,,,,,,,,,,,,,,,,,,,30,0,0,0,0,0,,,,FFFF$

Comando para borrar coordenadas en geocerca 1:
AT+GTPEO=gv55w,1,0,1,10,,,,,,,,,,,,,,,,,,,,,30,0,0,0,0,0,,,,FFFF$

Comando para crear una geocerca 0 cuadrada en El Tope (puse GEOID 0 por silas pero en realidad es 6):
AT+GTPEO=gv55w,0,3,1,4,-75.471595,10.378294,-75.471568,10.378721,-75.471450,10.378599,-75.471490,10.378165,30,0,0,0,0,0,,,,FFFF$

## XVU383

## XVU383 para pruebas de End Speeding

Se envió AT+GTSPD=gv55w,3,0,10,10,300,0,0,0,0,,,,,,,,,,,,FFFF$ el 23-05-2025 14:26

In [17]:
xvu383=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/xvu383.csv')
xvu383.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23423 entries, 0 to 23422
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  23423 non-null  object
 1   Message  23423 non-null  object
 2   Status   23423 non-null  object
dtypes: object(3)
memory usage: 549.1+ KB


In [18]:
xvu_383_gtspd=xvu383[xvu383['Message'].str.contains('GTSPD')]
xvu_383_gtspd
xvu_383_gtspd.to_csv('/home/jlhb1984/py-projects/data/geofences_on_device/xvu383_gtspd.csv')

## LA TORTU GTU922


Se establece ubicaciones históricas 22-05-2025 a 29-05-2025
R00A10V09, 102

In [19]:
gtu922=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlgtu922.csv',sep=';',encoding='latin1')
gtu922.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9190 entries, 0 to 9189
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     9190 non-null   int64  
 1   Unit         9190 non-null   object 
 2   Events       9190 non-null   object 
 3   Date         9190 non-null   object 
 4   Landmark     939 non-null    object 
 5   Address      9190 non-null   object 
 6   Speed        9190 non-null   int64  
 7   Temperature  0 non-null      float64
 8   Driver       9190 non-null   object 
 9   Latitude     9190 non-null   float64
 10  Longitude    9190 non-null   float64
dtypes: float64(3), int64(2), object(6)
memory usage: 789.9+ KB


In [20]:
gtu922_addresses=gtu922['Address'].value_counts()
gtu922_addresses

Address
13-47 Carrera 6                               491
1337 Carrera 6                                349
Av 50, Manizales, Caldas                      261
Via Principal, Manizales, Caldas, Colombia    162
Av 25, Manizales, Caldas                      128
                                             ... 
Cl67a 40-16, Manizales, Caldas                  1
Cl67a 40-81, Manizales, Caldas, Colombia        1
Kr41 67f-23, Manizales, Caldas                  1
Cl68 41-72, Manizales, Caldas                   1
Kr42 66c-76, Manizales, Caldas                  1
Name: count, Length: 3025, dtype: int64

In [21]:
gtu922['Address'].fillna('Empty',inplace=True)
gtu922
gtu922_landmarks=gtu922[gtu922['Address'].str.contains('Av 50, Manizales, Caldas')]
gtu922_landmarks

/tmp/ipykernel_33721/2425314806.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  gtu922['Address'].fillna('Empty',inplace=True)


Sequence             Unit Events                  Date Landmark  \
1150      8040  LA TORTU GTU922  Drive  Jun 19 2025 10:37 AM      NaN   
1370      7820  LA TORTU GTU922  Drive  Jun 19 2025 08:56 AM      NaN   
1371      7819  LA TORTU GTU922  Drive  Jun 19 2025 08:56 AM      NaN   
1372      7818  LA TORTU GTU922  Drive  Jun 19 2025 08:56 AM      NaN   
1375      7815  LA TORTU GTU922  Drive  Jun 19 2025 08:55 AM      NaN   
...        ...              ...    ...                   ...      ...   
7997      1193  LA TORTU GTU922  Drive  Jun 06 2025 11:26 AM      NaN   
8001      1189  LA TORTU GTU922  Drive  Jun 06 2025 11:24 AM      NaN   
8002      1188  LA TORTU GTU922  Drive  Jun 06 2025 11:23 AM      NaN   
8426       764  LA TORTU GTU922  Drive  Jun 04 2025 08:40 PM      NaN   
8428       762  LA TORTU GTU922  Drive  Jun 04 2025 08:39 PM      NaN   

                                 Address  Speed  Temperature  \
1150  Av 50, Manizales, Caldas, Colombia      9          NaN   
1370  Av 50, Manizales, Caldas, Colombia     22          NaN   
1371  Av 50, Manizales, Caldas, Colombia     20          NaN   
1372  Av 50, Manizales, Caldas, Colombia     22          NaN   
1375            Av 50, Manizales, Caldas     28          NaN   
...                                  ...    ...          ...   
7997            Av 50, Manizales, Caldas     41          NaN   
8001            Av 50, Manizales, Caldas     64          NaN   
8002            Av 50, Manizales, Caldas     65          NaN   
8426            Av 50, Manizales, Caldas     14          NaN   
8428            Av 50, Manizales, Caldas     32          NaN   

                                Driver  Latitude  Longitude  
1150  Cristian David, Valencia zuluaga  5.046108 -75.529494  
1370  Cristian David, Valencia zuluaga  5.034995 -75.465649  
1371  Cristian David, Valencia zuluaga  5.035137 -75.465555  
1372  Cristian David, Valencia zuluaga  5.035296 -75.465600  
1375  Cristian David, Valencia zuluaga  5.032781 -75.469482  
...                                ...       ...        ...  
7997  Cristian David, Valencia zuluaga  5.032815 -75.469485  
8001  Cristian David, Valencia zuluaga  5.038744 -75.484838  
8002  Cristian David, Valencia zuluaga  5.040525 -75.488751  
8426  Cristian David, Valencia zuluaga  5.046024 -75.529482  
8428  Cristian David, Valencia zuluaga  5.048053 -75.528977  

[349 rows x 11 columns]

In [22]:
gtu922['Address'].fillna('Empty',inplace=True)
gtu922
gtu922_landmarks=gtu922[gtu922['Address'].str.contains('Cra. 6 #13-47, Villamaría, Caldas, Colombia')]
gtu922_landmarks

Sequence             Unit       Events                  Date Landmark  \
8695       495  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:04 PM      NaN   
8696       494  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:04 PM      NaN   
8697       493  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:03 PM      NaN   
8698       492  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:02 PM      NaN   
8699       491  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:02 PM      NaN   
8700       490  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:01 PM      NaN   
8701       489  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:01 PM      NaN   
8702       488  LA TORTU GTU922  Idle - Stop  Jun 04 2025 04:00 PM      NaN   
8703       487  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:59 PM      NaN   
8704       486  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:58 PM      NaN   
8705       485  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:57 PM      NaN   
8706       484  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:56 PM      NaN   
8707       483  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:55 PM      NaN   
8708       482  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:54 PM      NaN   
8709       481  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:53 PM      NaN   
8710       480  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:52 PM      NaN   
8711       479  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:51 PM      NaN   
8712       478  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:50 PM      NaN   
8713       477  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:50 PM      NaN   
8714       476  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:49 PM      NaN   
8715       475  LA TORTU GTU922  Idle - Stop  Jun 04 2025 03:49 PM      NaN   
9152        38  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:27 PM      NaN   
9153        37  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:26 PM      NaN   
9154        36  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:26 PM      NaN   
9155        35  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:25 PM      NaN   
9156        34  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:25 PM      NaN   
9157        33  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:24 PM      NaN   
9158        32  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:23 PM      NaN   
9159        31  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:22 PM      NaN   
9160        30  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:21 PM      NaN   
9161        29  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:20 PM      NaN   
9162        28  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:19 PM      NaN   
9163        27  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:18 PM      NaN   
9164        26  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:17 PM      NaN   
9165        25  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:17 PM      NaN   
9166        24  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:16 PM      NaN   
9167        23  LA TORTU GTU922  Idle - Stop  Jun 03 2025 02:16 PM      NaN   

                                          Address  Speed  Temperature  \
8695  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8696  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8697  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8698  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8699  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8700  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8701  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8702  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8703  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8704  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8705  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8706  Cra. 6 #13-47, Villamaría, Caldas, Colombia      0          NaN   
8707  Cra. 6 #13-47, Villamaría, Caldas, Colombia     

In [23]:
gtu922_latitudes=gtu922['Latitude'].value_counts()
gtu922_latitudes

Latitude
5.042008    55
5.042743    55
4.856003    48
5.041959    47
5.042120    40
            ..
5.058442     1
5.055748     1
5.055384     1
5.055386     1
5.064611     1
Name: count, Length: 6347, dtype: int64

In [24]:
gtu922_longitude=gtu922['Longitude'].value_counts()
gtu922_longitude

Longitude
-75.510445    77
-75.655514    48
-75.510354    47
-75.510387    45
-75.511123    39
              ..
-75.509147     1
-75.506683     1
-75.505418     1
-75.503653     1
-75.522101     1
Name: count, Length: 6421, dtype: int64

## LA TORTU GTU206

Se establece ubicaciones históricas 22-05-2025 a 29-05-2025
R00A10V09, 102

In [25]:
gtu206=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlgtu206.csv',sep=';',encoding='latin1')
gtu206.info()

FileNotFoundError: [Errno 2] No such file or directory: '/home/jlhb1984/py-projects/data/geofences_on_device/hlgtu206.csv'

In [26]:
gtu206_addresses=gtu206['Address'].value_counts()
gtu206_addresses

NameError: name 'gtu206' is not defined

In [27]:
gtu206['Address'].fillna('Empty',inplace=True)
gtu206
gtu206_landmarks=gtu206[gtu206['Address'].str.contains('Kr6 13-39, Villamaria, Caldas, Colombia')]
gtu206_landmarks

NameError: name 'gtu206' is not defined

In [28]:
gtu206_gtspd=gtu206[gtu206['Message'].str.contains('GTSPD')]
gtu206_gtspd

NameError: name 'gtu206' is not defined

Pruebas GTU922:
gtu922:5.042872	-75.503047 AT+GTSPD=gv55w,3,0,10,10,300,0,0,0,0,,,,,,,,,,,,FFFF$ el 31-05-2025
gtu922:AT+GTGEO=gv55w,1,3,-75.51020900,5.04212400,299,0,0,0,0,0,0,0,0,,FFFF$     el 04-06-2025
gtu922:AT+GTGEO=gv55w,2,3,-75.51037600,5.04197700,310,0,0,0,0,0,0,0,0,,FFFF$
gtu922:AT+GTGEO=gv55w,1,3,-75.51020900,5.04212400,299,0,0,0,0,0,0,0,0,,FFFF$     el 21-06-2025 10:30am

Pruebas GTU206:
gtu206:5.042124,-75.510209 AT+GTGEO=gv55w,0,3,-75.510209,5.042124,100,30,0,0,0,0,0,0,0,,FFFF$ 290520251844
gtu206: AT+GTSPD=gv55w,3,0,10,10,300,0,0,0,0,,,,,,,,,,,,FFFF$ el 04-06-2025

In [29]:
gtu922_landmarks=gtu922[gtu922['Address'].str.contains('Av 50, Manizales, Caldas')]

## Nuevas pruebas 19062025

In [30]:
new_gio=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/Geofences InOut Report_19062025.csv')
new_gio.info()
new_gio.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8846 entries, 0 to 8845
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   textbox12       8846 non-null   object 
 1   UnitName        8846 non-null   object 
 2   Textbox3        0 non-null      float64
 3   EventTime       8846 non-null   object 
 4   GeofenceName    8846 non-null   object 
 5   GeofenceEvent1  8846 non-null   object 
 6   Textbox4        0 non-null      float64
dtypes: float64(2), object(5)
memory usage: 483.9+ KB


textbox12       UnitName  Textbox3                EventTime  \
0   Grupo:   Unidad:SRR902       NaN  31/05/2025 5:41:44 a.m.   
1   Grupo:   Unidad:SRR902       NaN  31/05/2025 5:42:34 a.m.   
2   Grupo:   Unidad:SRR902       NaN  31/05/2025 8:09:19 a.m.   
3   Grupo:   Unidad:SRR902       NaN  31/05/2025 8:44:34 a.m.   
4   Grupo:   Unidad:SRR902       NaN  31/05/2025 8:45:35 a.m.   

                                        GeofenceName GeofenceEvent1  Textbox4  
0  12.- Estbas del Bosque. : cartagena bosque car...        ENTRADA       NaN  
1  12.- Estbas del Bosque. : cartagena bosque car...         SALIDA       NaN  
2  6.- Seguro Social : cartagena traversal 54 car...        ENTRADA       NaN  
3  8.- Parque Espiritu del Manglar : cartagena ce...        ENTRADA       NaN  
4  8.- Parque Espiritu del Manglar : cartagena ce...         SALIDA       NaN

In [31]:
new_gio_geo1=new_gio[new_gio['GeofenceName'].str.contains('Turno')]
new_gio_geo1['UnitName'].value_counts()

UnitName
Unidad:UAP929     80
Unidad:XVO856     64
Unidad:UAN790     62
Unidad:UAO890     61
Unidad:XMB461     55
Unidad:XVO410     50
Unidad:UAR598     48
Unidad:XVN243     48
Unidad:XVO358     46
Unidad:TNF678     42
Unidad:TPY950     39
Unidad:TPQ091     37
Unidad:XVO853     37
Unidad:SRR661     37
Unidad:SPH808     36
Unidad:SLP174     36
Unidad:UAN646     35
Unidad:WHI947     34
Unidad:TPR 078    33
Unidad:TKI593     30
Unidad:UAQ054     29
Unidad:UAO041     29
Unidad:UAR494     28
Unidad:SLP442     28
Unidad:XVU383     27
Unidad:UAO056     25
Unidad:SRR902     23
Unidad:UAO940     21
Unidad:UAO861     19
Unidad:XVP649     18
Unidad:SJR801     16
Unidad:UAO586     13
Unidad:UAM325     10
Name: count, dtype: int64

## UAN790

In [32]:
uan790_geo1=new_gio_geo1[new_gio_geo1['UnitName'].str.contains('UAN790')]
uan790_geo1.info()
uan790_geo1.sort_values('EventTime')

<class 'pandas.core.frame.DataFrame'>
Index: 62 entries, 3721 to 4088
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   textbox12       62 non-null     object 
 1   UnitName        62 non-null     object 
 2   Textbox3        0 non-null      float64
 3   EventTime       62 non-null     object 
 4   GeofenceName    62 non-null     object 
 5   GeofenceEvent1  62 non-null     object 
 6   Textbox4        0 non-null      float64
dtypes: float64(2), object(5)
memory usage: 3.9+ KB


textbox12       UnitName  Textbox3                 EventTime  \
3927   Grupo:   Unidad:UAN790       NaN  10/06/2025 12:47:07 p.m.   
3928   Grupo:   Unidad:UAN790       NaN   10/06/2025 1:13:54 p.m.   
3910   Grupo:   Unidad:UAN790       NaN   10/06/2025 5:10:58 a.m.   
3911   Grupo:   Unidad:UAN790       NaN   10/06/2025 5:24:58 a.m.   
3921   Grupo:   Unidad:UAN790       NaN   10/06/2025 8:49:57 a.m.   
...        ...            ...       ...                       ...   
3897   Grupo:   Unidad:UAN790       NaN   9/06/2025 12:49:48 p.m.   
3904   Grupo:   Unidad:UAN790       NaN    9/06/2025 5:21:12 p.m.   
3905   Grupo:   Unidad:UAN790       NaN    9/06/2025 5:22:11 p.m.   
3887   Grupo:   Unidad:UAN790       NaN    9/06/2025 8:16:50 a.m.   
3888   Grupo:   Unidad:UAN790       NaN    9/06/2025 8:45:32 a.m.   

                                           GeofenceName GeofenceEvent1  \
3927  1. Turno de despacho : Calle 34 # 17 231 Carta...        ENTRADA   
3928  1. Turno de despacho : Calle 34 # 17 231 Carta...         SALIDA   
3910  1. Turno de despacho : Calle 34 # 17 231 Carta...        ENTRADA   
3911  1. Turno de despacho : Calle 34 # 17 231 Carta...         SALIDA   
3921  1. Turno de despacho : Calle 34 # 17 231 Carta...        ENTRADA   
...                                                 ...            ...   
3897  1. Turno de despacho : Calle 34 # 17 231 Carta...         SALIDA   
3904  1. Turno de despacho : Calle 34 # 17 231 Carta...        ENTRADA   
3905  1. Turno de despacho : Calle 34 # 17 231 Carta...         SALIDA   
3887  1. Turno de despacho : Calle 34 # 17 231 Carta...        ENTRADA   
3888  1. Turno de despacho : Calle 34 # 17 231 Carta...         SALIDA   

      Textbox4  
3927       NaN  
3928       NaN  
3910       NaN  
3911       NaN  
3921       NaN  
...        ...  
3897       NaN  
3904       NaN  
3905       NaN  
3887       NaN  
3888       NaN  

[62 rows x 7 columns]

## UAR598

19-06-2025

In [33]:
raw_data_uar598=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_uar598.csv',sep=';',encoding='latin1')
raw_data_uar598.info()
raw_data_uar598_gtgeo=raw_data_uar598[raw_data_uar598['Message'].str.contains('ACK')]
raw_data_uar598_gtgeo
#raw_data_uar598_gtgeo['Created'].sort_values(ascending=False)
#report=raw_data_uar598_gtgeo[raw_data_uar598_gtgeo['Created'].str.contains('2025-06-18')]
#report
#report.to_csv('/home/jlhb1984/py-projects/data/geofences_on_device/reportgtgeo_uar598.csv')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28055 entries, 0 to 28054
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  28055 non-null  object
 1   Message  28055 non-null  object
 2   Status   28055 non-null  object
dtypes: object(3)
memory usage: 657.7+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [34]:
hluar598=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hluar598.csv',sep=';',encoding='latin1')
hluar598.info()
hluar598.fillna('Nan',inplace=True)
hluar598_gtgeo_1=hluar598[hluar598['Pt. Referencia'].str.contains('Turno')]
hluar598_gtgeo_1.info()
hluar598_gtgeo_1.head(25)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26516 entries, 0 to 26515
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Secuencia       26516 non-null  int64  
 1   Unidad          26516 non-null  object 
 2   Eventos         26516 non-null  object 
 3   Fecha           26516 non-null  object 
 4   Pt. Referencia  5251 non-null   object 
 5   Dirección       26516 non-null  object 
 6   Velocidad       26516 non-null  int64  
 7   Temperatura     0 non-null      float64
 8   Conductor       26516 non-null  object 
 9   Latitud         26516 non-null  float64
 10  Longitud        26516 non-null  float64
dtypes: float64(3), int64(2), object(6)
memory usage: 2.2+ MB
<class 'pandas.core.frame.DataFrame'>
Index: 48 entries, 1467 to 26392
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Secuencia       48 non-null     int64  
 1   

/tmp/ipykernel_33721/1842775927.py:3: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'Nan' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  hluar598.fillna('Nan',inplace=True)


Secuencia  Unidad       Eventos                 Fecha  \
1467       25049  UAR598  Geofence OUT  Jun 18 2025 05:01 PM   
1476       25040  UAR598   Geofence IN  Jun 18 2025 04:54 PM   
1904       24612  UAR598  Geofence OUT  Jun 18 2025 12:52 PM   
1910       24606  UAR598   Geofence IN  Jun 18 2025 12:48 PM   
1937       24579  UAR598  Geofence OUT  Jun 18 2025 12:23 PM   
1940       24576  UAR598   Geofence IN  Jun 18 2025 12:22 PM   
2303       24213  UAR598  Geofence OUT  Jun 18 2025 08:48 AM   
2325       24191  UAR598   Geofence IN  Jun 18 2025 08:30 AM   
2678       23838  UAR598  Geofence OUT  Jun 18 2025 05:20 AM   
2697       23819  UAR598   Geofence IN  Jun 18 2025 05:04 AM   
3117       23399  UAR598  Geofence OUT  Jun 17 2025 05:19 PM   
3126       23390  UAR598   Geofence IN  Jun 17 2025 05:12 PM   
5863       20653  UAR598  Geofence OUT  Jun 16 2025 05:26 AM   
5887       20629  UAR598   Geofence IN  Jun 16 2025 05:05 AM   
6552       19964  UAR598   Geofence IN  Jun 14 2025 03:01 PM   
7012       19504  UAR598  Geofence OUT  Jun 14 2025 10:26 AM   
7015       19501  UAR598   Geofence IN  Jun 14 2025 10:25 AM   
8666       17850  UAR598  Geofence OUT  Jun 13 2025 05:50 AM   
8670       17846  UAR598   Geofence IN  Jun 13 2025 05:49 AM   
9415       17101  UAR598   Geofence IN  Jun 12 2025 12:34 PM   
10721      15795  UAR598  Geofence OUT  Jun 11 2025 04:28 PM   
10786      15730  UAR598   Geofence IN  Jun 11 2025 03:25 PM   
11194      15322  UAR598  Geofence OUT  Jun 11 2025 11:19 AM   
11197      15319  UAR598   Geofence IN  Jun 11 2025 11:18 AM   
11680      14836  UAR598  Geofence OUT  Jun 11 2025 06:23 AM   

             Pt. Referencia  \
1467   1. Turno de despacho   
1476   1. Turno de despacho   
1904   1. Turno de despacho   
1910   1. Turno de despacho   
1937   1. Turno de despacho   
1940   1. Turno de despacho   
2303   1. Turno de despacho   
2325   1. Turno de despacho   
2678   1. Turno de despacho   
2697   1. Turno de despacho   
3117   1. Turno de despacho   
3126   1. Turno de despacho   
5863   1. Turno de despacho   
5887   1. Turno de despacho   
6552   1. Turno de despacho   
7012   1. Turno de despacho   
7015   1. Turno de despacho   
8666   1. Turno de despacho   
8670   1. Turno de despacho   
9415   1. Turno de despacho   
10721  1. Turno de despacho   
10786  1. Turno de despacho   
11194  1. Turno de despacho   
11197  1. Turno de despacho   
11680  1. Turno de despacho   

                                               Dirección  Velocidad  \
1467                         Tr54-37, Cartagena, Bolivar          0   
1476                         Tr54-49, Cartagena, Bolivar          0   
1904                         Tr54-45, Cartagena, Bolivar         24   
1910               Tr54-50, Cartagena, Bolivar, Colombia          0   
1937   Tv. 54 #38, Cartagena de Indias, Provincia de ...          0   
1940   Tv. 54 #38, Cartagena de Indias, Provincia de ...          0   
2303                         Tr54-42, Cartagena, Bolivar         12   
2325                         Tr54-49, Cartagena, Bolivar          0   
2678                         Tr54-53, Cartagena, Bolivar          9   
2697                         Tr54-49, Cartagena, Bolivar          0   
3117                         Tr54-53, Cartagena, Bolivar          0   
3126                         Tr54-49, Cartagena, Bolivar          0   
5863                         Tr54-52, Cartagena, Bolivar         12   
5887                         Tr54-49, Cartagena, Bolivar          0   
6552   Tv. 54 #38, Cartagena de Indias, Provincia de ...          9   
7012                         Tr54-22, Cartagena, Bolivar          0   
7015                         Tr54-49, Cartagena, Bolivar          0   
8666                         Tr54-14, Cartagena, Bolivar          0   
8670                         Tr54-49, Cartagena, Bolivar          9   
9415                         Tr54-49, Cartagena, Bolivar          0   
10721                        Tr54-45, Cartag

In [35]:
uar598_new_rd=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/uar598_new_rd.csv',sep=';',encoding='latin1')
uar598_new_rd.info()
uar598_new_rd_gtgeo=uar598_new_rd[uar598_new_rd['Message'].str.contains('ACK:GTGEO')]
uar598_new_rd_gtgeo

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72211 entries, 0 to 72210
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  72211 non-null  object
 1   Message  72211 non-null  object
 2   Status   72211 non-null  object
dtypes: object(3)
memory usage: 1.7+ MB


Created  \
8187   2025-05-09 15:58:54.812   
11076  2025-05-11 13:00:53.592   
20725  2025-05-17 16:53:22.136   
20729  2025-05-17 16:55:03.623   
20734  2025-05-17 16:56:25.358   
20742  2025-05-17 16:58:45.267   
20743  2025-05-17 16:58:45.827   
20744  2025-05-17 16:58:46.310   
20745  2025-05-17 16:58:46.928   
20746  2025-05-17 16:58:47.548   
20747  2025-05-17 16:58:48.089   
20748  2025-05-17 16:58:48.609   
20749  2025-05-17 16:58:49.150   
20779  2025-05-17 17:12:31.088   
20780  2025-05-17 17:12:51.429   
20784  2025-05-17 17:14:11.813   
21243  2025-05-17 21:39:32.439   
21249  2025-05-17 21:42:13.169   
21255  2025-05-17 21:43:13.162   
21258  2025-05-17 21:44:34.159   
21262  2025-05-17 21:45:35.410   
21266  2025-05-17 21:46:35.364   
21269  2025-05-17 21:47:14.324   
21273  2025-05-17 21:48:17.250   
21277  2025-05-17 21:49:18.716   
21280  2025-05-17 21:50:17.419   
21284  2025-05-17 21:51:17.714   
21287  2025-05-17 21:52:18.922   
21290  2025-05-17 21:53:18.961   

                                                 Message  \
8187   +ACK:GTGEO,500700,868574042858816,,1,FFFF,2025...   
11076  +ACK:GTGEO,500700,868574042858816,,0,FFFF,2025...   
20725  +ACK:GTGEO,500700,868574042858816,,0,FFFF,2025...   
20729  +ACK:GTGEO,500700,868574042858816,,5,FFFF,2025...   
20734  +ACK:GTGEO,500700,868574042858816,,2,FFFF,2025...   
20742  +ACK:GTGEO,500700,868574042858816,,12,FFFF,202...   
20743  +ACK:GTGEO,500700,868574042858816,,7,FFFF,2025...   
20744  +ACK:GTGEO,500700,868574042858816,,11,FFFF,202...   
20745  +ACK:GTGEO,500700,868574042858816,,1,FFFF,2025...   
20746  +ACK:GTGEO,500700,868574042858816,,9,FFFF,2025...   
20747  +ACK:GTGEO,500700,868574042858816,,3,FFFF,2025...   
20748  +ACK:GTGEO,500700,868574042858816,,10,FFFF,202...   
20749  +ACK:GTGEO,500700,868574042858816,,4,FFFF,2025...   
20779  +ACK:GTGEO,500700,868574042858816,,7,FFFF,2025...   
20780  +ACK:GTGEO,500700,868574042858816,,8,FFFF,2025...   
20784  +ACK:GTGEO,500700,868574042858816,,8,FFFF,2025...   
21243  +ACK:GTGEO,500700,868574042858816,,0,FFFF,2025...   
21249  +ACK:GTGEO,500700,868574042858816,,1,FFFF,2025...   
21255  +ACK:GTGEO,500700,868574042858816,,2,FFFF,2025...   
21258  +ACK:GTGEO,500700,868574042858816,,3,FFFF,2025...   
21262  +ACK:GTGEO,500700,868574042858816,,4,FFFF,2025...   
21266  +ACK:GTGEO,500700,868574042858816,,5,FFFF,2025...   
21269  +ACK:GTGEO,500700,868574042858816,,6,FFFF,2025...   
21273  +ACK:GTGEO,500700,868574042858816,,7,FFFF,2025...   
21277  +ACK:GTGEO,500700,868574042858816,,8,FFFF,2025...   
21280  +ACK:GTGEO,500700,868574042858816,,9,FFFF,2025...   
21284  +ACK:GTGEO,500700,868574042858816,,10,FFFF,202...   
21287  +ACK:GTGEO,500700,868574042858816,,11,FFFF,202...   
21290  +ACK:GTGEO,500700,868574042858816,,12,FFFF,202...   

                                                  Status  
8187   0x2B41434B3A475447454F2C3530303730302C38363835...  
11076  0x2B41434B3A475447454F2C3530303730302C38363835...  
20725  0x2B41434B3A475447454F2C3530303730302C38363835...  
20729  0x2B41434B3A475447454F2C3530303730302C38363835...  
20734  0x2B41434B3A475447454F2C3530303730302C38363835...  
20742  0x2B41434B3A475447454F2C3530303730302C38363835...  
20743  0x2B41434B3A475447454F2C3530303730302C38363835...  
20744  0x2B41434B3A475447454F2C3530303730302C38363835...  
20745  0x2B41434B3A475447454F2C3530303730302C38363835...  
20746  0x2B41434B3A475447454F2C3530303730302C38363835...  
20747  0x2B41434B3A475447454F2C3530303730302C38363835...  
20748  0x2B41434B3A475447454F2C3530303730302C38363835...  
20749  0x2B41434B3A475447454F2C3530303730302C38363835...  
20779  0x2B41434B3A475447454F2C3530303730302C38363835...  
20780  0x2B41434B3A475447454F2C3530303730302C38363835...  
20784  0x2B41434B3A475447454F2C3530303730302C38363835...  
21243  0x2B41434B3A475447454F2C3530303730302C38363835...  
21249  0x2B41434B3A475447454F2C3530303730302C38363835...  
21255  0x2B41434B3A475447454F2C3530303730302C38363835...  
21258  0x2B41434B3A47

In [36]:
raw_data_uan790=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_uan790.csv',sep=';',encoding='latin1')
raw_data_uan790.info()
raw_data_uan790_gtgeo=raw_data_uan790[raw_data_uan790['Message'].str.contains('GTGEO')]
raw_data_uan790_gtgeo.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2978 entries, 0 to 2977
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  2978 non-null   object
 1   Message  2978 non-null   object
 2   Status   2978 non-null   object
dtypes: object(3)
memory usage: 69.9+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [37]:
raw_data_sjr801=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_sjr801.csv',sep=';',encoding='latin1')
raw_data_sjr801.info()
raw_data_sjr801_gtgeo=raw_data_sjr801[raw_data_sjr801['Message'].str.contains('GTGEO')]
raw_data_sjr801_gtgeo.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20729 entries, 0 to 20728
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  20729 non-null  object
 1   Message  20729 non-null  object
 2   Status   20729 non-null  object
dtypes: object(3)
memory usage: 486.0+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [38]:
raw_data_slp442=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_slp442.csv',sep=';',encoding='latin1')
raw_data_slp442.info()
raw_data_slp442_gtgeo=raw_data_slp442[raw_data_slp442['Message'].str.contains('GTGEO')]
raw_data_slp442_gtgeo.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19732 entries, 0 to 19731
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  19732 non-null  object
 1   Message  19732 non-null  object
 2   Status   19732 non-null  object
dtypes: object(3)
memory usage: 462.6+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [39]:
raw_data_sph808=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_sph808.csv',sep=';',encoding='latin1')
raw_data_sph808.info()
raw_data_sph808_gtgeo=raw_data_sph808[raw_data_sph808['Message'].str.contains('GTGEO')]
raw_data_sph808_gtgeo.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22748 entries, 0 to 22747
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  22748 non-null  object
 1   Message  22748 non-null  object
 2   Status   22748 non-null  object
dtypes: object(3)
memory usage: 533.3+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

In [40]:
raw_data_spr902=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_sph808.csv',sep=';',encoding='latin1')
raw_data_spr902.info()
raw_data_spr902_gtgeo=raw_data_spr902[raw_data_sph808['Message'].str.contains('GTGEO')]
raw_data_spr902_gtgeo.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22748 entries, 0 to 22747
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  22748 non-null  object
 1   Message  22748 non-null  object
 2   Status   22748 non-null  object
dtypes: object(3)
memory usage: 533.3+ KB


Empty DataFrame
Columns: [Created, Message, Status]
Index: []

## Nuevas pruebas GTU922. 21-06-2025.

In [41]:
gtu922_210625=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/gtu922.csv')
gtu922_210625.info()
gtu922_210625_gtgeo=gtu922_210625[gtu922_210625['Message'].str.contains('ACK')]
gtu922_210625_gtgeo

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 942 entries, 0 to 941
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  942 non-null    object
 1   Message  942 non-null    object
 2   Status   942 non-null    object
dtypes: object(3)
memory usage: 22.2+ KB


Created  \
280  2025-06-04 21:15:00.659   

                                               Message  \
280  +ACK:GTRTO,500501,868574042763735,,RTL,FFFF,20...   

                                                Status  
280  0x2B41434B3A475452544F2C3530303530312C38363835...

Se obtuvo el archivo hlgtu922 en la página en inglés.

In [42]:
gtu_922_hl=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlgtu922.csv',sep=';',encoding='latin1')
gtu_922_hl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9190 entries, 0 to 9189
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     9190 non-null   int64  
 1   Unit         9190 non-null   object 
 2   Events       9190 non-null   object 
 3   Date         9190 non-null   object 
 4   Landmark     939 non-null    object 
 5   Address      9190 non-null   object 
 6   Speed        9190 non-null   int64  
 7   Temperature  0 non-null      float64
 8   Driver       9190 non-null   object 
 9   Latitude     9190 non-null   float64
 10  Longitude    9190 non-null   float64
dtypes: float64(3), int64(2), object(6)
memory usage: 789.9+ KB


Landmarks:

In [43]:
gtu922_landmarks=gtu_922_hl[gtu_922_hl['Events'].str.contains('Landmark')]
gtu922_landmarks.info()
gtu922_landmarks.to_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlgtu922_landmark.csv')

<class 'pandas.core.frame.DataFrame'>
Index: 41 entries, 84 to 8100
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     41 non-null     int64  
 1   Unit         41 non-null     object 
 2   Events       41 non-null     object 
 3   Date         41 non-null     object 
 4   Landmark     41 non-null     object 
 5   Address      41 non-null     object 
 6   Speed        41 non-null     int64  
 7   Temperature  0 non-null      float64
 8   Driver       41 non-null     object 
 9   Latitude     41 non-null     float64
 10  Longitude    41 non-null     float64
dtypes: float64(3), int64(2), object(6)
memory usage: 3.8+ KB


Teleamigo:

In [44]:
telea_0406_0506=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/teleamigo.csv',sep=';',encoding='latin1')
telea_0406_0506.info()
telea_0406_0506_message=telea_0406_0506[telea_0406_0506['text'].str.contains('GTGEO')]
telea_0406_0506_message

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3819 entries, 0 to 3818
Data columns (total 17 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   requestType         3819 non-null   object 
 1   inbound             0 non-null      float64
 2   outbound            3819 non-null   object 
 3   sendAt              3819 non-null   object 
 4   doneAt              3819 non-null   object 
 5   sender              3819 non-null   object 
 6   destination         3791 non-null   float64
 7   externalMessageId   3819 non-null   object 
 8   status              3819 non-null   object 
 9   text                3819 non-null   object 
 10  error               3819 non-null   object 
 11  errorGroup          3819 non-null   object 
 12  seenAt              3819 non-null   object 
 13  ivrResponses        0 non-null      float64
 14  ivrMappedResponses  0 non-null      float64
 15  applicationId       3819 non-null   object 
 16  entity

requestType  inbound outbound               sendAt               doneAt  \
501          SMS      NaN        X  05/06/2025 16:00:17  05/06/2025 16:00:37   
2322         SMS      NaN        X  04/06/2025 16:50:15  04/06/2025 16:50:18   

      sender   destination       externalMessageId     status  \
501   890505  5.791036e+11  4491572178104335743159  Delivered   
2322  890505  5.791036e+11  4490738153104335752115  Delivered   

                                                   text              error  \
501   AT+GTGEO=gv55w,2,3,-75.51037600,5.04197700,310...  No Error (code 0)   
2322  AT+GTGEO=gv55w,1,3,-75.51020900,5.04212400,299...  No Error (code 0)   

     errorGroup seenAt  ivrResponses  ivrMappedResponses applicationId  \
501   No Errors     --           NaN                 NaN       default   
2322  No Errors     --           NaN                 NaN       default   

      entityId  
501        NaN  
2322       NaN

## Conclusiones 21-06-2025:
1. Solo está funcionando la detección de landmarks en el dispositio UAR598 porque se le envió el comando desde CRM y con Teleamigo operativo, es muy probable que cuando se envió los comandos con el módulo de Carlos, había fallas en Teleamigo.
2. Cuando la unidad recibe el comando exitosamente, debe haber un ACK, que solo llegó a la unidad UAR598, cuando se envió desde el CRM en la etapa de pruebas antes de pasar el caso a desarrollo.
3. Se procede a buscar otra unidad de pruebas, diferente a la unidad UAR598 y con una cuenta que no esté en mora.
4. La placa GTU206 de Harold mantiene en estado de parqueado, por lo tanto no es una buena opción para pruebas.

In [45]:
new_units=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/ubd21062025.csv',sep=';',encoding='latin1')
new_units.info()
queclink_units=new_units[new_units['Unit Type'].str.contains('Queclink GV55W')]
queclink_units_reporting=queclink_units[['Unit Type','Unit ID','Carrier','Phone Number','Last Event Date','Company','Unit']]
queclink_units_reporting[queclink_units_reporting['Last Event Date'].str.contains('6/21')]

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12067 entries, 0 to 12066
Data columns (total 28 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Dealer               12067 non-null  object
 1   Company              12067 non-null  object
 2   VIP                  12067 non-null  bool  
 3   FEIN                 12067 non-null  object
 4   DNI                  1789 non-null   object
 5   Peripherals          12067 non-null  object
 6   Service Name         10373 non-null  object
 7   Owner                12067 non-null  object
 8   Unit ID              12067 non-null  object
 9   Unit                 12067 non-null  object
 10  Unit Type            12067 non-null  object
 11  Carrier              11450 non-null  object
 12  Monthly Fee          12067 non-null  object
 13  Billing Group        628 non-null    object
 14  Billing Period       11979 non-null  object
 15  Last Srv Paid Until  12067 non-null  object
 16  Serv

Unit Type Unit ID   Carrier Phone Number        Last Event Date  \
3      Queclink GV55W    12PU  Claro CO   9104051709  6/21/2025 11:25:26 AM   
5      Queclink GV55W    13IL  Claro CO   9103544886  6/21/2025 10:46:02 AM   
6      Queclink GV55W    13IR  Claro CO   9103544880  6/21/2025 11:26:53 AM   
7      Queclink GV55W    14CF  Claro CO   9103427365  6/21/2025 11:07:57 AM   
8      Queclink GV55W    14CK  Claro CO   9103429852  6/21/2025 11:26:50 AM   
...               ...     ...       ...          ...                    ...   
12061  Queclink GV55W    790Z  Claro CO   9103418722  6/21/2025 11:27:01 AM   
12062  Queclink GV55W    J844  Claro CO   9103381460  6/21/2025 11:25:39 AM   
12063  Queclink GV55W    L263  Claro CO   9103386472  6/21/2025 11:27:20 AM   
12064  Queclink GV55W    S533  Claro CO   9103389068  6/21/2025 11:13:01 AM   
12065  Queclink GV55W    S540  Claro CO   9103446102  6/21/2025 11:27:15 AM   

                  Company              Unit  
3      4G4 INGENIERIA SAS            HSR591  
5      4G4 INGENIERIA SAS     Komatsu PC210  
6      4G4 INGENIERIA SAS  Bobcat S570 2017  
7      4G4 INGENIERIA SAS     Komatsu PC130  
8      4G4 INGENIERIA SAS       VOLVO EC35D  
...                   ...               ...  
12061  ZONA LOGISTICA SAS            LKO873  
12062  ZONA LOGISTICA SAS            LKO878  
12063  ZONA LOGISTICA SAS            LKO874  
12064  ZONA LOGISTICA SAS            LKO880  
12065  ZONA LOGISTICA SAS            LKO877  

[2870 rows x 7 columns]

## HSR591

R00A10V09 0102

Envío de coordenada 3.181024,-75.220932 por CRM, a 0 segundos AT+GTGEO=gv55w,1,3,-75.220932,3.181024,50,0,0,0,0,0,0,0,0,,FFFF$ 21-06-2025 11:55am
Recibió ACK: +ACK:GTGEO,500501,868574042783121,,1,FFFF,20250621165502,5C5E$

Envío de coordenada 3.182345,-75.220595 por CRM, a 30 segundos AT+GTGEO=gv55w,2,3,-75.220595,3.182345,150,30,0,0,0,0,0,0,0,,FFFF$
Recibió ACK: +ACK:GTGEO,500501,868574042783121,,2,FFFF,20250621174603,5CA3$

Ambos fueron verificados como recepción exitosa en Teleamigo!

In [46]:
hlhsr591=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlhsr591.csv',sep=';',encoding='latin1')
hlhsr591.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     513 non-null    int64  
 1   Unit         513 non-null    object 
 2   Events       513 non-null    object 
 3   Date         513 non-null    object 
 4   Landmark     0 non-null      float64
 5   Address      513 non-null    object 
 6   Speed        513 non-null    int64  
 7   Temperature  0 non-null      float64
 8   Driver       0 non-null      float64
 9   Latitude     513 non-null    float64
 10  Longitude    513 non-null    float64
dtypes: float64(5), int64(2), object(4)
memory usage: 44.2+ KB


In [47]:
hlhsr591_Address=hlhsr591['Address'].value_counts()
hlhsr591_Address

Address
Villavieja, Tello, Huila, Colombia                             82
Neiva - Baraya- Villa Vieja-4, Villavieja, Huila, Colombia     39
5QMH+FQ Villavieja, Huila, Colombia                            26
Neiva - Baraya- Villa Vieja-39, Villavieja, Huila, Colombia    20
5QMH+CR Villavieja, Huila, Colombia                            16
                                                               ..
Cl18 52-4, Neiva, Huila, Colombia                               1
Cra. 52, Neiva, Neiva, Huila, Colombia                          1
Cl. 8 #41-516 a 41-612, Neiva, Neiva, Huila, Colombia           1
C. 8 # 46-62, Neiva, Huila, Colombia                            1
C. 8 #46-18, Neiva, Huila, Colombia                             1
Name: count, Length: 186, dtype: int64

In [48]:
hlhsr591_lat_long_1=hlhsr591[hlhsr591['Address'].str.contains('Neiva - Baraya- Villa Vieja-4, Villavieja, Huila, Colombia')]
hlhsr591_lat_long_1

Sequence    Unit                        Events                  Date  \
9         504  HSR591                   Idle - Stop  Jun 21 2025 11:17 AM   
20        493  HSR591            Ignición Encendida  Jun 21 2025 11:09 AM   
21        492  HSR591                   Idle - Stop  Jun 21 2025 11:09 AM   
22        491  HSR591                        Parked  Jun 21 2025 11:09 AM   
23        490  HSR591                        Parked  Jun 21 2025 11:09 AM   
24        489  HSR591  Desplazándose - Ign. apagada  Jun 21 2025 11:09 AM   
25        488  HSR591              Ignición Apagada  Jun 21 2025 11:09 AM   
26        487  HSR591                         Drive  Jun 21 2025 11:08 AM   
27        486  HSR591                   Idle - Stop  Jun 21 2025 11:08 AM   
28        485  HSR591            Ignición Encendida  Jun 21 2025 11:08 AM   
29        484  HSR591                        Parked  Jun 21 2025 10:45 AM   
30        483  HSR591                        Parked  Jun 21 2025 10:44 AM   
31        482  HSR591              Ignición Apagada  Jun 21 2025 10:44 AM   
32        481  HSR591  Desplazándose - Ign. apagada  Jun 21 2025 10:44 AM   
59        454  HSR591                   Idle - Stop  Jun 21 2025 10:31 AM   
60        453  HSR591                   Idle - Stop  Jun 21 2025 10:30 AM   
61        452  HSR591                   Idle - Stop  Jun 21 2025 10:29 AM   
114       399  HSR591            Ignición Encendida  Jun 21 2025 09:50 AM   
115       398  HSR591                   Idle - Stop  Jun 21 2025 09:50 AM   
116       397  HSR591                        Parked  Jun 21 2025 09:49 AM   
117       396  HSR591                        Parked  Jun 21 2025 09:48 AM   
118       395  HSR591              Ignición Apagada  Jun 21 2025 09:48 AM   
119       394  HSR591  Desplazándose - Ign. apagada  Jun 21 2025 09:48 AM   
120       393  HSR591                   Idle - Stop  Jun 21 2025 09:48 AM   
122       391  HSR591                   Idle - Stop  Jun 21 2025 09:46 AM   
181       332  HSR591                         Drive  Jun 21 2025 09:14 AM   
202       311  HSR591                   Idle - Stop  Jun 21 2025 08:59 AM   
203       310  HSR591                   Idle - Stop  Jun 21 2025 08:59 AM   
204       309  HSR591                   Idle - Stop  Jun 21 2025 08:59 AM   
205       308  HSR591                   Idle - Stop  Jun 21 2025 08:58 AM   
206       307  HSR591                   Idle - Stop  Jun 21 2025 08:57 AM   
220       293  HSR591                   Idle - Stop  Jun 21 2025 08:46 AM   
221       292  HSR591                   Idle - Stop  Jun 21 2025 08:45 AM   
222       291  HSR591                   Idle - Stop  Jun 21 2025 08:44 AM   
223       290  HSR591                   Idle - Stop  Jun 21 2025 08:43 AM   
224       289  HSR591                   Idle - Stop  Jun 21 2025 08:43 AM   
227       286  HSR591                   Idle - Stop  Jun 21 2025 08:42 AM   
228       285  HSR591                   Idle - Stop  Jun 21 2025 08:41 AM   
229       284  HSR591                   Idle - Stop  Jun 21 2025 08:40 AM   

     Landmark                                            Address  Speed  \
9         NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
20        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
21        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
22        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
23        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
24        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
25        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
26        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...     17   
27        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
28        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
29        NaN  Neiva - Baraya- Villa Vieja-4, Villavieja, Hui...      0   
30        NaN  Neiva - B

## HFX606

R00A12V03 0102

Envío de coordenada 2 veces 4.645272,-74.198755 desde CRM. AT+GTGEO=gv55w,2,3,-74.198755,4.645272,50,30,0,0,0,0,0,0,0,,FFFF$ 21-06-2025 12:28pm 

Recepción ACK: +ACK:GTGEO,500700,868574042894548,,2,FFFF,20250621172858,43A4$

Ambos fueron verificados como recepción exitosa en Teleamigo!

In [49]:
hlhfx606=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hlhfx606.csv',sep=';',encoding='latin1')
hlhfx606.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6303 entries, 0 to 6302
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     6303 non-null   int64  
 1   Unit         6303 non-null   object 
 2   Events       6303 non-null   object 
 3   Date         6303 non-null   object 
 4   Landmark     619 non-null    object 
 5   Address      6303 non-null   object 
 6   Speed        6303 non-null   int64  
 7   Temperature  0 non-null      float64
 8   Driver       0 non-null      float64
 9   Latitude     6303 non-null   float64
 10  Longitude    6303 non-null   float64
dtypes: float64(4), int64(2), object(5)
memory usage: 541.8+ KB


In [50]:
hlhfx606_Address=hlhfx606['Address'].value_counts()
hlhfx606_Address

Address
67P7JRW2+3F                                                                760
Carrera 106 A                                                              582
67P7JRW2+4F                                                                539
67P7JRW2+5C                                                                264
67P7JRW2+2G                                                                219
                                                                          ... 
Cl10e 7-24, Soacha, Cundinamarca, Colombia                                   1
Cl. 11 #9 Este-2 a 9 Este-42, Soacha, Soacha, Cundinamarca, Colombia         1
Cl11e 11c-34, Soacha, Cundinamarca, Colombia                                 1
Tv. 12 Este #11-8 a 11-34, Soacha, Soacha, Cundinamarca, Colombia            1
Transversal 6 Este #7-26 a 7-76, Soacha, Soacha, Cundinamarca, Colombia      1
Name: count, Length: 861, dtype: int64

In [51]:
hlhfx606_lat_long_1=hlhfx606[hlhfx606['Address'].str.contains('3F')]
hlhfx606_lat_long_1

Sequence    Unit       Events                  Date Landmark  \
593       5710  HFX606  Idle - Stop  Jun 19 2025 01:40 PM      NaN   
594       5709  HFX606  Idle - Stop  Jun 19 2025 01:40 PM      NaN   
595       5708  HFX606  Idle - Stop  Jun 19 2025 01:40 PM      NaN   
596       5707  HFX606  Idle - Stop  Jun 19 2025 01:39 PM      NaN   
597       5706  HFX606  Idle - Stop  Jun 19 2025 01:38 PM      NaN   
...        ...     ...          ...                   ...      ...   
6259        44  HFX606  Idle - Stop  Jun 01 2025 08:23 AM      NaN   
6260        43  HFX606  Idle - Stop  Jun 01 2025 08:22 AM      NaN   
6261        42  HFX606  Idle - Stop  Jun 01 2025 08:21 AM      NaN   
6262        41  HFX606  Idle - Stop  Jun 01 2025 08:20 AM      NaN   
6263        40  HFX606  Idle - Stop  Jun 01 2025 08:19 AM      NaN   

          Address  Speed  Temperature  Driver  Latitude  Longitude  
593   67P7JRW2+3F      0          NaN     NaN  4.645272 -74.198755  
594   67P7JRW2+3F      0          NaN     NaN  4.645245 -74.198701  
595   67P7JRW2+3F      0          NaN     NaN  4.645245 -74.198701  
596   67P7JRW2+3F      0          NaN     NaN  4.645245 -74.198701  
597   67P7JRW2+3F      0          NaN     NaN  4.645245 -74.198701  
...           ...    ...          ...     ...       ...        ...  
6259  67P7JRW2+3F      0          NaN     NaN  4.645243 -74.198766  
6260  67P7JRW2+3F      0          NaN     NaN  4.645243 -74.198766  
6261  67P7JRW2+3F      0          NaN     NaN  4.645243 -74.198766  
6262  67P7JRW2+3F      0          NaN     NaN  4.645243 -74.198766  
6263  67P7JRW2+3F      0          NaN     NaN  4.645243 -74.198766  

[764 rows x 11 columns]

## LKO873

R00A10V09, 0102

In [52]:
hllko873=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hllko873.csv',sep=';',encoding='latin1')
hllko873.info()
hllko873.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17057 entries, 0 to 17056
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     17057 non-null  int64  
 1   Unit         17057 non-null  object 
 2   Events       17057 non-null  object 
 3   Date         17057 non-null  object 
 4   Landmark     0 non-null      float64
 5   Address      17057 non-null  object 
 6   Speed        17057 non-null  int64  
 7   Temperature  0 non-null      float64
 8   Driver       0 non-null      float64
 9   Latitude     17057 non-null  float64
 10  Longitude    17057 non-null  float64
dtypes: float64(5), int64(2), object(4)
memory usage: 1.4+ MB


Sequence    Unit       Events                  Date  Landmark  \
0     17057  LKO873  Idle - Stop  Jun 21 2025 01:17 PM       NaN   
1     17056  LKO873  Idle - Stop  Jun 21 2025 01:17 PM       NaN   
2     17055  LKO873  Idle - Stop  Jun 21 2025 01:16 PM       NaN   
3     17054  LKO873        Drive  Jun 21 2025 01:16 PM       NaN   
4     17053  LKO873  Idle - Stop  Jun 21 2025 01:16 PM       NaN   

                               Address  Speed  Temperature  Driver  Latitude  \
0  Kr105f 67d-90, Bogota, Cundinamarca      0          NaN     NaN  4.701036   
1  Kr105f 67d-90, Bogota, Cundinamarca      0          NaN     NaN  4.701036   
2  Kr105f 67d-90, Bogota, Cundinamarca      0          NaN     NaN  4.701071   
3  Kr105f 67f-45, Bogota, Cundinamarca      9          NaN     NaN  4.701139   
4   Cl68 105d-95, Bogota, Cundinamarca      0          NaN     NaN  4.701173   

   Longitude  
0 -74.124339  
1 -74.124339  
2 -74.124313  
3 -74.124215  
4 -74.124148

In [53]:
hllko873['Date'].sort_values(ascending=False)

35       Jun 21 2025 12:59 PM
37       Jun 21 2025 12:58 PM
36       Jun 21 2025 12:58 PM
38       Jun 21 2025 12:57 PM
40       Jun 21 2025 12:56 PM
                 ...         
17053    Jun 01 2025 03:43 AM
17042    Jun 01 2025 02:45 PM
17054    Jun 01 2025 02:43 AM
17043    Jun 01 2025 01:44 PM
17055    Jun 01 2025 01:42 AM
Name: Date, Length: 17057, dtype: object

In [54]:
hllko873_Address=hllko873['Address'].value_counts()
hllko873_Address

Address
Cl22d 127-37, Bogota, Cundinamarca, Colombia                    583
Carrera 132 # 22A-1 a 22A-99, Bogotá, Cundinamarca, Colombia    260
Cl22d 127-45, Bogota, Cundinamarca, Colombia                    168
Kr132 22a-20, Bogota, Cundinamarca, Colombia                    152
Cl22d 127-39, Bogota, Cundinamarca, Colombia                    132
                                                               ... 
Cl63-64, Bogota, Cundinamarca                                     1
Tr85 63-82, Bogota, Cundinamarca                                  1
Cl63b 80a-53, Bogota, Cundinamarca                                1
Cl63b 79-73, Bogota, Cundinamarca                                 1
Cl26 99-34, Bogota, Cundinamarca, Colombia                        1
Name: count, Length: 6477, dtype: int64

In [55]:
hllko873_lat_long_1=hllko873[hllko873['Address'].str.contains('Cl22d 127-37, Bogota, Cundinamarca, Colombia')]
hllko873_lat_long_1

Sequence    Unit       Events                  Date  Landmark  \
2587      14470  LKO873  Idle - Stop  Jun 19 2025 08:56 AM       NaN   
2588      14469  LKO873  Idle - Stop  Jun 19 2025 08:55 AM       NaN   
2607      14450  LKO873  Idle - Stop  Jun 19 2025 08:45 AM       NaN   
2708      14349  LKO873  Idle - Stop  Jun 19 2025 06:14 AM       NaN   
2709      14348  LKO873  Ignition ON  Jun 19 2025 06:14 AM       NaN   
...         ...     ...          ...                   ...       ...   
17052         5  LKO873       Parked  Jun 01 2025 04:43 AM       NaN   
17053         4  LKO873       Parked  Jun 01 2025 03:43 AM       NaN   
17054         3  LKO873       Parked  Jun 01 2025 02:43 AM       NaN   
17055         2  LKO873       Parked  Jun 01 2025 01:42 AM       NaN   
17056         1  LKO873       Parked  Jun 01 2025 12:42 AM       NaN   

                                            Address  Speed  Temperature  \
2587   Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
2588   Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
2607   Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
2708   Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
2709   Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
...                                             ...    ...          ...   
17052  Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
17053  Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
17054  Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
17055  Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   
17056  Cl22d 127-37, Bogota, Cundinamarca, Colombia      0          NaN   

       Driver  Latitude  Longitude  
2587      NaN  4.695235 -74.156173  
2588      NaN  4.695292 -74.156112  
2607      NaN  4.695292 -74.156107  
2708      NaN  4.695231 -74.156118  
2709      NaN  4.695231 -74.156118  
...       ...       ...        ...  
17052     NaN  4.695258 -74.156036  
17053     NaN  4.695258 -74.156036  
17054     NaN  4.695258 -74.156036  
17055     NaN  4.695258 -74.156036  
17056     NaN  4.695258 -74.156036  

[583 rows x 11 columns]

Se hizo 3 intentos desde el módulo de Carlos pero no se recibió ACK:

AT+GTGEO=gv55w,1,3,-74.15611200,4.69529200,310,30,0,0,0,0,0,0,0,,FFFF$

Se hizo un nuevo envío de este mismo comando desde CRM sin éxito.
 
Se verificó recepción exitosa en Teleamigo.

Se hizo un nuevo intento usando Manage tool para generar:
AT+GTGEO=gv55w,1,3,-74.156112,4.695292,50,0,0,0,0,0,0,0,0,,FFFF$ Se envió desde CRM y se generó ACK:+ACK:GTGEO,500501,868574042430053,,1,FFFF,20250621205934,0AD8$

Se debe intentar solo con 6 cifras decimales.

Actualizar caso en JIRA.

https://gpstrackit.atlassian.net/browse/TSO-3420

Nuevas pruebas 24-06-2025.
Unidades de Zona Logistica SAS

In [56]:
zona_logistica_units=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/zona_logisitica_units.csv',sep=';',encoding='latin1')
zona_logistica_units.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53 entries, 0 to 52
Data columns (total 47 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       53 non-null     int64  
 1   Owner                    53 non-null     object 
 2   Unit ID                  53 non-null     object 
 3   Unit                     53 non-null     object 
 4   Peripherals              41 non-null     object 
 5   Reporting Status         0 non-null      float64
 6   Show Until               0 non-null      float64
 7   Billing Group            0 non-null      float64
 8   Last Event Date          53 non-null     object 
 9   Monthly Fee              53 non-null     int64  
 10  Bill To                  0 non-null      float64
 11  Assigned To              53 non-null     object 
 12  Linked Mode              53 non-null     int64  
 13  Paired GPS               0 non-null      float64
 14  Paired Camera            0 n

In [57]:
zona_logistica_units_queclink=zona_logistica_units[zona_logistica_units['Unit Type'].str.contains('Queclink')]
zona_logistica_units_queclink[['Unit','Unit ID','Unit Type','IMEI','Last Event Date']]

Unit Unit ID       Unit Type             IMEI Last Event Date
9   LKO880    S533  Queclink GV55W  868574042430335      25/06/2025
10  LKO879    252H  Queclink GV55W  868574042434865      25/06/2025
11  LKO878    J844  Queclink GV55W  868574042447461      25/06/2025
12  LKO877    S540  Queclink GV55W  868574042413703      25/06/2025
13  LKO876    163F  Queclink GV55W  868574042429733      25/06/2025
14  LKO875    102R  Queclink GV55W  868574042448980      25/06/2025
15  LKO874    L263  Queclink GV55W  868574042435128      25/06/2025
16  LKO873    790Z  Queclink GV55W  868574042430053      25/06/2025
18  GTX653    0JF8  Queclink GV55W  868574042956263      25/06/2025
45  ESP852    0KA0  Queclink GV55W  868574041343141      25/06/2025

Después de exponer el caso a desarrollo, se hizo la modificación a 6 cifras decimales. Inicio pruebas con la unidad LKO873 porque tiene acceso a configuración.

In [58]:
hllko873_lat_long_2=hllko873[hllko873['Address'].str.contains('Carrera 132 # 22A-1 a 22A-99, Bogotá, Cundinamarca, Colombia')]
hllko873_lat_long_2
#print('Latitudes y longitudes')
hllko873_lat_2=hllko873_lat_long_2['Latitude'].value_counts()
hllko873_long_2=hllko873_lat_long_2['Longitude'].value_counts()
hllko873_lat_2.head(5)


Latitude
4.698290    44
4.698248    24
4.698398    20
4.698292    19
4.698254    18
Name: count, dtype: int64

In [59]:
hllko873_long_2.head(5)

Longitude
-74.161505    50
-74.161555    24
-74.161576    23
-74.161560    21
-74.161559    20
Name: count, dtype: int64

Tramas de prueba:
Landmark3: AT+GTGEO=gv55w,3,3,-74.161555,4.698248,50,30,0,0,0,0,0,0,0,,FFFF$ x
Landmark4: AT+GTGEO=gv55w,4,3,-74.161576,4.698398,50,30,0,0,0,0,0,0,0,,FFFF$
Landmark5: AT+GTGEO=gv55w,5,3,-74.161560,4.698292,50,30,0,0,0,0,0,0,0,,FFFF$
Landmark6: AT+GTGEO=gv55w,6,3,-74.161559,4.698254,50,30,0,0,0,0,0,0,0,,FFFF$ 

## LKO878

R00A10V09, 0102

Se envió AT+GTGEO=gv55w,1,3,-74.161440,4.698221,50000,30,0,0,0,0,0,0,0,,FFFF$
con ACK +ACK:GTGEO,500501,868574042447461,,1,FFFF,20250626135033,FA83$ y +RESP:GTGEO,500501,868574042447461,,,10,1,1,0.0,345,43.0,-74.850652,10.932023,20250626135056,0732,0101,3AA3,BB81806,04,12,56774.6,20250626135057,FA84$
Aquí nos dimos cuenta del error en el radio, pusimos 50 y se generó 50000

Se usó Landmark3: AT+GTGEO=gv55w,3,3,-74.161555,4.698248,50,30,0,0,0,0,0,0,0,,FFFF$ 01-07-2025 12:18 Falló porque Teleamigo no está operativo.

In [60]:
lko878_260625=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/lko878.csv',sep=';',encoding='latin1')
lko878_260625.info()
print(lko878_260625.loc[0:1])
print(lko878_260625.loc[5464:5465])
lko878_260625_GTGEO=lko878_260625[lko878_260625['Message'].str.contains('GTGEO')]
lko878_260625_GTGEO


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5465 entries, 0 to 5464
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  5465 non-null   object
 1   Message  5465 non-null   object
 2   Status   5465 non-null   object
dtypes: object(3)
memory usage: 128.2+ KB
                   Created                                            Message  \
0  2025-06-26 00:00:09.738  +RESP:GTFRI,500501,868574042447461,,,10,1,1,73...   
1  2025-06-26 00:01:10.122  +RESP:GTFRI,500501,868574042447461,,,10,1,1,75...   

                                              Status  
0  0x2B524553503A47544652492C3530303530312C383638...  
1  0x2B524553503A47544652492C3530303530312C383638...  
                      Created  \
5464  2025-07-01 20:10:36.458   

                                                Message  \
5464  +RESP:GTFRI,500501,868574042447461,,,11,1,1,6....   

                                                 Status  
5464  0x2B5245

Created  \
663  2025-06-26 13:50:33.844   
664  2025-06-26 13:50:56.575   

                                               Message  \
663  +ACK:GTGEO,500501,868574042447461,,1,FFFF,2025...   
664  +RESP:GTGEO,500501,868574042447461,,,10,1,1,0....   

                                                Status  
663  0x2B41434B3A475447454F2C3530303530312C38363835...  
664  0x2B524553503A475447454F2C3530303530312C383638...

08-07-2025:
Se ejecutó resend command + save para LKO878 08-07-2025 12:54

AT+GTGEO=gv55w,1,3,-74.161555,4.698248,59,30,0,0,0,0,0,0,0,,FFFF$
+ACK:GTGEO,500501,868574042447461,,1,FFFF,20250708180022,2549$
+RESP:GTGEO,500501,868574042447461,,,10,1,1,0.0,109,56.2,-74.802674,10.904695,20250708180056,0732,0101,3A9A,BB81BC6,04,12,58946.8,20250708180055,254B$

In [61]:
raw_data_lko878_08072025=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_lko878_08072025.csv',sep=';',encoding='latin1')
gtgeo_events=raw_data_lko878_08072025[raw_data_lko878_08072025['Message'].str.contains('GTGEO')]
gtgeo_events

Message  \
1841  +ACK:GTGEO,500501,868574042447461,,1,FFFF,2025...   
1843  +RESP:GTGEO,500501,868574042447461,,,10,1,1,0....   

                                                 Status  
1841  0x2B41434B3A475447454F2C3530303530312C38363835...  
1843  0x2B524553503A475447454F2C3530303530312C383638...

Análisis de coordenadas para establecer las mas visitadas:

In [62]:
hllko878=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hllko8780106202511072025.csv',sep=';',encoding='latin1')
hllko878.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31529 entries, 0 to 31528
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Sequence     31529 non-null  int64  
 1   Unit         31529 non-null  object 
 2   Events       31529 non-null  object 
 3   Date         31529 non-null  object 
 4   Landmark     1 non-null      object 
 5   Address      31529 non-null  object 
 6   Speed        31529 non-null  int64  
 7   Temperature  0 non-null      float64
 8   Driver       31529 non-null  object 
 9   Latitude     31529 non-null  float64
 10  Longitude    31529 non-null  float64
dtypes: float64(3), int64(2), object(6)
memory usage: 2.6+ MB


In [63]:
hllko878['Latitude'].value_counts()

Latitude
10.986733    249
10.931861    213
10.931903    154
10.943217    143
10.931915    141
            ... 
10.932573      1
10.932751      1
10.921895      1
10.921972      1
10.922216      1
Name: count, Length: 15308, dtype: int64

In [64]:
hllko878['Longitude'].value_counts()

Longitude
-74.850429    226
-74.850301    155
-74.767904    149
-74.844132    135
-74.850445    129
             ... 
-75.486506      1
-75.485294      1
-75.485309      1
-75.508716      1
-75.485176      1
Name: count, Length: 15463, dtype: int64

11072025: Se creó landmark TSO trial 110720251037 10.931861 y -74.850429, radio 1Km:
AT+GTGEO=gv55w,2,3,-74.850429,10.931861,999,30,0,0,0,0,0,0,0,,FFFF$
+ACK:GTGEO,500501,868574042447461,,2,FFFF,20250711155032,2FDD$
+RESP:GTGEO,500501,868574042447461,,,21,1,1,0.0,325,38.7,-74.853211,10.937871,20250711155059,0732,0101,3AA3,BB817B7,04,12,59203.8,20250711155059,2FDE$

14-07-2025:
Se generó rdlko8781007_1407


In [65]:
raw_data_lko878=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/rawdatalko8781007_1407.csv',sep=';',encoding='latin1')
raw_data_lko878.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3656 entries, 0 to 3655
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  3656 non-null   object
 1   Message  3656 non-null   object
 2   Status   3656 non-null   object
dtypes: object(3)
memory usage: 85.8+ KB


In [66]:
gtgeo_events_lko878=raw_data_lko878[raw_data_lko878['Message'].str.contains('GTGEO')]
gtgeo_events_lko878.info()
gtgeo_events_lko878.to_csv('/home/jlhb1984/py-projects/data/geofences_on_device/raw_data_report_gtgeo.csv')

<class 'pandas.core.frame.DataFrame'>
Index: 18 entries, 1772 to 3373
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  18 non-null     object
 1   Message  18 non-null     object
 2   Status   18 non-null     object
dtypes: object(3)
memory usage: 576.0+ bytes


In [67]:
hllko878_10071407=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/hllko8781007_1407.csv',sep=';',encoding='latin1')
hllko878_10071407.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3212 entries, 0 to 3211
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Secuencia       3212 non-null   int64  
 1   Unidad          3212 non-null   object 
 2   Eventos         3212 non-null   object 
 3   Fecha           3212 non-null   object 
 4   Pt. Referencia  376 non-null    object 
 5   Dirección       3212 non-null   object 
 6   Velocidad       3212 non-null   int64  
 7   Temperatura     0 non-null      float64
 8   Conductor       3212 non-null   object 
 9   Latitud         3212 non-null   float64
 10  Longitud        3212 non-null   float64
dtypes: float64(3), int64(2), object(6)
memory usage: 276.2+ KB


In [68]:
hllko878_10071407['Eventos'].value_counts()

Eventos
Inmóvil - Detenido              2066
Conduciendo                      952
Estacionado                      112
Ignición Encendida                24
Ignición Apagada                  23
Entrando a Landmark               14
Saliendo de Landmark              11
Desplazándose - Ign. apagada      10
Name: count, dtype: int64

In [69]:
hllko878_10071407['Pt. Referencia'].value_counts()

Pt. Referencia
TSO trial 110720251037    376
Name: count, dtype: int64

In [70]:
hllko878_10071407['Eventos'].value_counts()

Eventos
Inmóvil - Detenido              2066
Conduciendo                      952
Estacionado                      112
Ignición Encendida                24
Ignición Apagada                  23
Entrando a Landmark               14
Saliendo de Landmark              11
Desplazándose - Ign. apagada      10
Name: count, dtype: int64

In [71]:
hllko878_10071407.fillna('Empty',inplace=True)
pt_ref=hllko878_10071407[hllko878_10071407['Eventos'].str.contains('Landmark')]
pt_ref

/tmp/ipykernel_33721/3151258754.py:1: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value 'Empty' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  hllko878_10071407.fillna('Empty',inplace=True)


Secuencia  Unidad               Eventos                 Fecha  \
292        2920  LKO878  Saliendo de Landmark  Jul 14 2025 06:42 AM   
465        2747  LKO878   Entrando a Landmark  Jul 12 2025 01:18 PM   
467        2745  LKO878   Entrando a Landmark  Jul 12 2025 01:18 PM   
548        2664  LKO878  Saliendo de Landmark  Jul 12 2025 12:23 PM   
552        2660  LKO878   Entrando a Landmark  Jul 12 2025 12:21 PM   
592        2620  LKO878  Saliendo de Landmark  Jul 12 2025 12:01 PM   
594        2618  LKO878   Entrando a Landmark  Jul 12 2025 11:59 AM   
657        2555  LKO878  Saliendo de Landmark  Jul 12 2025 10:36 AM   
691        2521  LKO878   Entrando a Landmark  Jul 12 2025 10:16 AM   
693        2519  LKO878   Entrando a Landmark  Jul 12 2025 10:16 AM   
725        2487  LKO878  Saliendo de Landmark  Jul 12 2025 08:46 AM   
727        2485  LKO878   Entrando a Landmark  Jul 12 2025 08:45 AM   
736        2476  LKO878  Saliendo de Landmark  Jul 12 2025 08:39 AM   
763        2449  LKO878   Entrando a Landmark  Jul 12 2025 08:27 AM   
819        2393  LKO878  Saliendo de Landmark  Jul 12 2025 07:49 AM   
822        2390  LKO878  Saliendo de Landmark  Jul 12 2025 07:47 AM   
824        2388  LKO878   Entrando a Landmark  Jul 12 2025 07:47 AM   
904        2308  LKO878   Entrando a Landmark  Jul 11 2025 07:00 PM   
906        2306  LKO878   Entrando a Landmark  Jul 11 2025 06:59 PM   
908        2304  LKO878  Saliendo de Landmark  Jul 11 2025 06:59 PM   
1703       1509  LKO878  Saliendo de Landmark  Jul 11 2025 11:08 AM   
1705       1507  LKO878   Entrando a Landmark  Jul 11 2025 11:07 AM   
1706       1506  LKO878  Saliendo de Landmark  Jul 11 2025 11:06 AM   
1727       1485  LKO878   Entrando a Landmark  Jul 11 2025 10:50 AM   
1735       1477  LKO878   Entrando a Landmark  Jul 11 2025 10:45 AM   

              Pt. Referencia  \
292   TSO trial 110720251037   
465   TSO trial 110720251037   
467   TSO trial 110720251037   
548   TSO trial 110720251037   
552   TSO trial 110720251037   
592   TSO trial 110720251037   
594   TSO trial 110720251037   
657   TSO trial 110720251037   
691   TSO trial 110720251037   
693   TSO trial 110720251037   
725   TSO trial 110720251037   
727   TSO trial 110720251037   
736   TSO trial 110720251037   
763   TSO trial 110720251037   
819   TSO trial 110720251037   
822   TSO trial 110720251037   
824   TSO trial 110720251037   
904   TSO trial 110720251037   
906   TSO trial 110720251037   
908   TSO trial 110720251037   
1703  TSO trial 110720251037   
1705  TSO trial 110720251037   
1706  TSO trial 110720251037   
1727  TSO trial 110720251037   
1735  TSO trial 110720251037   

                                              Dirección  Velocidad  \
292   Km 1 de la vía Troncal del Caribe  - Baranoa -...         67   
465                         Av 90-75, Galapa, Atlantico         54   
467   Km 2 de la vía Troncal del Caribe  - Baranoa -...         56   
548   Km 1 de la vía Troncal del Caribe  - Baranoa -...         57   
552   Km 3 de la vía Troncal del Caribe  - Baranoa -...         54   
592   Km 3 de la vía Troncal del Caribe  - Baranoa -...         51   
594   Km 2 de la vía Troncal del Caribe  - Baranoa -...         53   
657   Km 1 de la vía Troncal del Caribe  - Baranoa -...         59   
691   Km 3 de la vía Troncal del Caribe  - Baranoa -...         57   
693   Km 3 de la vía Troncal del Caribe  - Baranoa -...         57   
725   Km 3 de la vía Troncal del Caribe  - Baranoa -...         74   
727   Km 2 de la vía Troncal del Caribe  - Baranoa -...         72   
736   Km 1 de la vía Troncal del Caribe  - Baranoa -...         72   
763   Km 2 de la vía Troncal del Caribe  - Baranoa -...         72   
819   Km 1 de la vía Troncal del Caribe  - Baranoa -...          0   
822   Km 2 de la vía Troncal del Caribe  - Baranoa -...         65   
824   Km 2 de la vía Troncal del Caribe  - Baranoa -...         45   
904   Km 2 de la vía Troncal del Caribe  - Baranoa -...         12

In [4]:
customer_units_0108=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/cu01082025.csv',sep=';',encoding='latin1')
customer_units_0108.info()
customer_units_0108.head(5)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49 entries, 0 to 48
Data columns (total 47 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       49 non-null     int64  
 1   Owner                    49 non-null     object 
 2   Unit ID                  49 non-null     object 
 3   Unit                     49 non-null     object 
 4   Peripherals              0 non-null      float64
 5   Reporting Status         0 non-null      float64
 6   Show Until               0 non-null      float64
 7   Billing Group            0 non-null      float64
 8   Last Event Date          49 non-null     object 
 9   Monthly Fee              49 non-null     int64  
 10  Bill To                  0 non-null      float64
 11  Assigned To              49 non-null     object 
 12  Linked Mode              49 non-null     int64  
 13  Paired GPS               0 non-null      float64
 14  Paired Camera            0 n

ID    Owner Unit ID    Unit  Peripherals  Reporting Status  \
0  1105405  Renting    15D6  XVU952          NaN               NaN   
1  1105802  Renting    15O4  XVU383          NaN               NaN   
2  1081724  Renting    0LIF  XVU236          NaN               NaN   
3  1104816  Renting    14N8  XVP649          NaN               NaN   
4  1106038  Renting    15UO  XVO856          NaN               NaN   

   Show Until  Billing Group Last Event Date  Monthly Fee  ...  \
0         NaN            NaN      17/03/2025        49220  ...   
1         NaN            NaN       1/08/2025        49220  ...   
2         NaN            NaN      22/11/2024        49220  ...   
3         NaN            NaN       1/08/2025        49220  ...   
4         NaN            NaN       1/08/2025        49220  ...   

      Assigned to Company IFTA  History Months  Comments  \
0    11/4/2024 9:14:55 PM    N               3       NaN   
1  11/11/2024 10:48:29 PM    N               3       NaN   
2  12/19/2024 10:25:10 AM    N               3       NaN   
3   11/9/2024 11:49:49 AM    N               3       NaN   
4  12/15/2024 12:01:45 PM    N               3       NaN   

        Warranty Expires Script JBOX Profile Contract Type  Subscription  \
0   9/4/2025 12:00:00 AM    NaN          NaN           NaN           NaN   
1   9/4/2025 12:00:00 AM    NaN          NaN           NaN           NaN   
2   9/9/2021 12:00:00 AM    NaN          NaN           NaN           NaN   
3  6/12/2025 12:00:00 AM    NaN          NaN           NaN           NaN   
4  9/17/2025 12:00:00 AM    NaN          NaN           NaN           NaN   

  Firmware  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  

[5 rows x 47 columns]

In [14]:
units_god_01082025=customer_units_0108[customer_units_0108['Last Event Date'].str.contains('1/08/2025')]
units_god_01082025.info()
units_god_01082025['Unit'].value_counts()
#c=input("Digite la placa: ")
#units_god_01082025_placa=units_god_01082025[units_god_01082025['Unit'].str.contains(c)]
#units_god_01082025_placa

<class 'pandas.core.frame.DataFrame'>
Index: 40 entries, 1 to 48
Data columns (total 47 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       40 non-null     int64  
 1   Owner                    40 non-null     object 
 2   Unit ID                  40 non-null     object 
 3   Unit                     40 non-null     object 
 4   Peripherals              0 non-null      float64
 5   Reporting Status         0 non-null      float64
 6   Show Until               0 non-null      float64
 7   Billing Group            0 non-null      float64
 8   Last Event Date          40 non-null     object 
 9   Monthly Fee              40 non-null     int64  
 10  Bill To                  0 non-null      float64
 11  Assigned To              40 non-null     object 
 12  Linked Mode              40 non-null     int64  
 13  Paired GPS               0 non-null      float64
 14  Paired Camera            0 non-nu

Unit
XVU383     1
XVP649     1
XVO856     1
XVO853     1
XVO410     1
XVO358     1
XVN243     1
XMB461     1
WHK429     1
WHI947     1
WGM533     1
WGL390     1
VCG310     1
UAR598     1
UAQ054     1
UAP929     1
UAP281     1
UAO969     1
UAO940     1
UAO890     1
UAO861     1
UAO586     1
UAO056     1
UAO041     1
UAN790     1
UAN646     1
TRD944     1
TPY950     1
TPU785     1
TPR 078    1
TPQ091     1
TNF678     1
TKI593     1
TKI052     1
TES586     1
SRR902     1
SRR661     1
SPH808     1
SLP174     1
SJR801     1
Name: count, dtype: int64

In [7]:
raw_data_sjr801_01082025=pd.read_csv('/home/jlhb1984/py-projects/data/geofences_on_device/rdsjr801_01082025.csv',sep=';',encoding='latin1')
raw_data_sjr801_01082025.info()
ack_gtgeo=raw_data_sjr801_01082025[raw_data_sjr801_01082025['Message'].str.contains('ACK')]
ack_gtgeo

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 719 entries, 0 to 718
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Created  719 non-null    object
 1   Message  719 non-null    object
 2   Status   719 non-null    object
dtypes: object(3)
memory usage: 17.0+ KB


Created  \
405  2025-08-01 14:09:23.470   
433  2025-08-01 14:40:19.723   
434  2025-08-01 15:10:41.748   
436  2025-08-01 15:40:27.471   
437  2025-08-01 16:09:33.371   
439  2025-08-01 16:40:33.287   
440  2025-08-01 17:09:37.172   
442  2025-08-01 17:39:57.312   
443  2025-08-01 18:10:40.558   
445  2025-08-01 18:40:17.311   
446  2025-08-01 19:10:35.982   
503  2025-08-01 19:40:21.819   

                                               Message  \
405  +ACK:GTGEO,500700,868574042924162,,1,FFFF,2025...   
433  +ACK:GTGEO,500700,868574042924162,,2,FFFF,2025...   
434  +ACK:GTGEO,500700,868574042924162,,11,FFFF,202...   
436  +ACK:GTGEO,500700,868574042924162,,12,FFFF,202...   
437  +ACK:GTGEO,500700,868574042924162,,3,FFFF,2025...   
439  +ACK:GTGEO,500700,868574042924162,,4,FFFF,2025...   
440  +ACK:GTGEO,500700,868574042924162,,5,FFFF,2025...   
442  +ACK:GTGEO,500700,868574042924162,,6,FFFF,2025...   
443  +ACK:GTGEO,500700,868574042924162,,7,FFFF,2025...   
445  +ACK:GTGEO,500700,868574042924162,,8,FFFF,2025...   
446  +ACK:GTGEO,500700,868574042924162,,9,FFFF,2025...   
503  +ACK:GTGEO,500700,868574042924162,,10,FFFF,202...   

                                                Status  
405  0x2B41434B3A475447454F2C3530303730302C38363835...  
433  0x2B41434B3A475447454F2C3530303730302C38363835...  
434  0x2B41434B3A475447454F2C3530303730302C38363835...  
436  0x2B41434B3A475447454F2C3530303730302C38363835...  
437  0x2B41434B3A475447454F2C3530303730302C38363835...  
439  0x2B41434B3A475447454F2C3530303730302C38363835...  
440  0x2B41434B3A475447454F2C3530303730302C38363835...  
442  0x2B41434B3A475447454F2C3530303730302C38363835...  
443  0x2B41434B3A475447454F2C3530303730302C38363835...  
445  0x2B41434B3A475447454F2C3530303730302C38363835...  
446  0x2B41434B3A475447454F2C3530303730302C38363835...  
503  0x2B41434B3A475447454F2C3530303730302C38363835...